applications/tari_conduit/src/grpc_conversions/blocks.rs

```diff
@@ -0,0 +1,291 @@
+// Copyright 2020. The Tari Project
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use crate::grpc;
+use prost_types::Timestamp;
+use std::convert::{TryFrom, TryInto};
+use tari_core::{
+    blocks::{Block, BlockHeader, NewBlockHeaderTemplate, NewBlockTemplate},
+    chain_storage::HistoricalBlock,
+    proof_of_work::{Difficulty, PowAlgorithm, ProofOfWork},
+    transactions::types::BlindingFactor,
+};
+use tari_crypto::tari_utilities::{epoch_time::EpochTime, ByteArray, Hashable};
+
+/// Utility function that converts a `chrono::DateTime` to a `prost::Timestamp`
+pub fn datetime_to_timestamp(datetime: EpochTime) -> Timestamp {
+    Timestamp {
+        seconds: datetime.as_u64() as i64,
+        nanos: 0,
+    }
+}
+
+pub(crate) fn timestamp_to_datetime(timestamp: Timestamp) -> EpochTime {
+    (timestamp.seconds as u64).into()
+}
+
+impl From<HistoricalBlock> for grpc::HistoricalBlock {
```

Comment:
I dont think we are using this?

Reply:
This file is copied over from grpc_conversions in the base node, transactions.rs as well.

Reply:
They needed a minor modification to get it to work